syntaxes/gdb.tmLanguage.json

```diff
@@ -398,6 +447,51 @@
 						}
 					]
 				},
+				{
+					"name": "meta.block.while.gdb",
+					"begin": "^\\s*(while)?\\s+(.*)\\s*$",
+					"beginCaptures": {
+						"1": { "name": "keyword.control.command.gdb" },
+						"2": { "name": "keyword.control.condition.gdb" }
+					},
+					"end": "^\\s*(end)\\s*$",
+					"endCaptures": {
+						"1": { "name": "keyword.control.end.gdb"}
+					},
+					"patterns": [
+						{ "include": "#expression" }
+					]
+				},
+				{
+					"name": "meta.block.command.gdb",
+					"begin": "^\\s*(define(?:-prefix)?\\s+comm(?:a(?:n(?:d)?)?)?)\\s+(\\w+)\\s*$",
+					"beginCaptures": {
+						"1": { "name": "keyword.control.command.gdb" },
+						"2": { "name": "variable.other.variable.gdb" }
+					},
+					"end": "^\\s*(end)\\s*$",
+					"endCaptures": {
+						"1": { "name": "keyword.control.end.gdb"}
+					},
+					"patterns": [
+						{ "include": "#expression" }
+					]
+				},
+				{
+					"name": "meta.block.document.gdb",
+					"begin": "^\\s*(doc(?:ument))\\s+(\\w+)\\s*$",
```

Comment:
'ument' can be split in consecutive letters, and also you missed the ? operator at the end of it, it can be omited
Proposing `(doc(?:u(?:m(?:e(?:n(?:t)?)?)?)?)?)`

Reply:
taken